block/retriever.go

```diff
@@ -36,6 +37,22 @@ func (m *Manager) ApplyBatchFromSL(slBatch *settlement.Batch) error {
 				continue
 			}
 
+			if block.GetRevision() != m.State.GetRevision() {
+
+				rollapp, err := m.SLClient.GetRollapp()
+				if err != nil {
+					return err
+				}
+				if m.shouldStopNode(rollapp, m.State.GetRevision()) {
```

Comment:
not sure why we check it inside the loop.
a batch has single revision
this can be checked at the start

Reply:
because revision is not serialized in the batch, therefore it needs to be obtained from the block in retrieval

Reply:
so it should be added in `convertStateInfoToResultRetrieveBatch`